### DISCUSSION TITLE
Ticker stops after one tick if Update() called tea.batch()

### DISCUSSION THREAD

**Discussion by Dentrax** [charmbracelet/bubbletea#551] [Q&A]
Hey,

I was trying [pager](https://github.com/charmbracelet/bubbletea/blob/master/examples/pager/main.go) example. Added a ticker func into `Init` funciton as follows:

```
func (m model) Init() tea.Cmd {
	return tea.Tick(5*time.Second, func(t time.Time) tea.Msg {
		return tickMsg(t)
	})
}
```

My `tickMsg` is:

```
var tickMsg time.Time
```

In the update function, I've added a new case:

```
case tickMsg:
		if r.ready {
			rrr := rand.Int()
			r.viewport.SetContent(strconv.Itoa(rrr))
		}
```

I'm not sure if it's a bug or not, but when I run the program, it waits 5-sec then prints a random number. And ticker stops immediately. I also tried to add that ticker in `Update()` func:

```
    ...
    r.viewport, cmd = r.viewport.Update(msg)
	cmds = append(cmds, cmd)
	cmds = append(cmds, r.tick)

	return r, tea.Batch(cmds...)
```

But this does not work either. Try to press random keys and resize the terminal. It seems my `ticker` duplicates and started to print random number in every _miliseconds_. After a while, terminal freezes.

How can I make it work? I want to use a viewport with a ticker to render scheduled. Thanks!

**Comment by bashbunni** [comment]
Hey @Dentrax I'd try returning a `tea.Tick` when you receive a `tickMsg` similar to [this](https://github.com/charmbracelet/bubbletea/blob/6547773a3f677c91cd261f307d7f5162973d1170/examples/fullscreen/main.go) example. Let me know if that fixes the issue or not

**Comment by inkel** [comment]
@Dentrax I believe that what you're trying to achieve is what [`tea.Every`](https://pkg.go.dev/github.com/charmbracelet/bubbletea#Every) gives, and not `Tick`. Note that even the documentation says it only runs once:

> Beginners' note: Tick sends a single message and won't automatically dispatch messages at an interval. To do that, you'll want to return another Tick command after receiving your tick message.

I believe that using the word `interval` in `Tick` documentation is confusing.

**Comment by Dentrax** [comment]
Thanks for responses! I couldn't find a bandwidth to test these, I will update this issue once I get into it! 😊
